Add note to installation instructions that Windows compatibility is not supported

_Issue [JP-1474](https://jira.stsci.edu/browse/JP-1474) was created on JIRA by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe):_

We should add a note to the installation instructions (and mabye ReadTheDocs/JDox?) that, while the software should install and run for Windows, it is not currently supported. This question came up in the DMSWG meeting today.

Based on the results of #4993, there are 163 unit test failures in `jwst` when run on Windows.

https://travis-ci.org/github/spacetelescope/jwst/jobs/692239993#L76133-L76304

A number of these come via `crds`, which also does not support Windows.

```
>               raise CrdsDownloadError(
                    "Failed caching mapping files:", str(exc)) from exc
E               crds.core.exceptions.CrdsDownloadError: Failed caching mapping files: Error fetching data for 'jwst_system_datalvl_0002.rmap' at CRDS server 'https://jwst-crds.stsci.edu' with mode 'http' : [WinError 3] The system cannot find the path specified: 'C:\\Users\\travis\\AppData\\Local\\Temp\\crds_cache\\mappings\\jwst'

c:\python38\lib\site-packages\crds\core\heavy_client.py:354: CrdsDownloadError
```

And others come from the assumption that we are running on a unix system, but can probably easily be fixed in the future.

```
>       log.log.info("Hostname: {0}".format(os.uname()[1]))
E       AttributeError: module 'os' has no attribute 'uname'
```

So I suspect a hard "We do not support Windows" would be the correct message now.

--- a/README.md
+++ b/README.md
@@ -8,7 +8,9 @@
 
 ![STScI Logo](docs/_static/stsci_logo.png)
 
-**JWST requires Python 3.6 or above and a C compiler for dependencies**.
+**JWST requires Python 3.6 or above and a C compiler for dependencies.**
+
+**Linux and MacOS platforms are tested and supported.  Windows is not currently supported.**
 
 
 ## Installation

